content/en/blog/falcosidekick-kubeless.md

```diff
@@ -0,0 +1,329 @@
+
+```shell
+kubectl get pods -n kubeless
+
+NAME                                          READY   STATUS    RESTARTS   AGE
+kubeless-controller-manager-99459cb67-tb99d   3/3     Running   3          3d14h
+delete-pod-d6f98f6dd-cw228                    1/1     Running   0          2m52s
+```
+```shell
+kubectl get svc -n kubeless
+
+NAME         TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)          AGE
+delete-pod   ClusterIP   10.43.211.201   <none>        8080/TCP         4m38s
+```
+
+## Test our function
+
+We start by creating a dumb pod:
+
+```shell
+kubectl run alpine -n default --image=alpine --restart='Never' -- sh -c "sleep 600"
+```
+```shell
+kubectl get pods -n default
+NAME     READY   STATUS    RESTARTS   AGE
+alpine   1/1     Running   0          9s
+```
+
+Let's run a *shell* command inside and see what happens:
+
+```shell
+kubectl exec -i --tty alpine -n default -- sh -c "uptime"
+
+23:44:25 up 1 day, 19:11,  load average: 0.87, 0.77, 0.77
+```
+
+As expected we got the result of our command, but, if get the status of the pod now:
+
+```shell
+kubectl get pods -n default
+NAME     READY   STATUS        RESTARTS   AGE
+alpine   1/1     Terminating   0          103s
+```
+
+💥 **It has been terminated** 💥
+
+We can now check the logs of components.
+
+For `Falco`:
+```bash
+kubectl logs daemonset/falco -n falco
+
+{"output":"23:39:44.834631763: Notice A shell was spawned in a container with an attached terminal (user=root user_loginuid=-1 k8s.ns=default k8s.pod=alpine container=5892b41bcf46 shell=sh parent=<NA> cmdline=sh terminal=34817 container_id=5892b41bcf46 image=<NA>) k8s.ns=default k8s.pod=alpine container=5892b41bcf46","priority":"Notice","rule":"Terminal shell in container","time":"2021-01-14T23:39:44.834631763Z", "output_fields": {"container.id":"5892b41bcf46","container.image.repository":null,"evt.time":1610667584834631763,"k8s.ns.name":"default","k8s.pod.name":"alpine","proc.cmdline":"sh","proc.name":"sh","proc.pname":null,"proc.tty":34817,"user.loginuid":-1,"user.name":"root"}}
+```
+
+For `Falcosidekick`:
+```shell
+kubectl logs deployment/falcosidekick -n falco
+
+2021/01/14 23:39:45 [INFO]  : Kubeless - Post OK (200)
+2021/01/14 23:39:45 [INFO]  : Kubeless - Function Response : 
+2021/01/14 23:39:45 [INFO]  : Kubeless - Call Function "delete-pod" OK
+```
+
+*(Notice, the function returns nothing, this is why the message log is empty)*
+
+For `delete-pod` function:
+```shell
+kubectl logs deployment/delete-pod -n kubeless
+
+10.42.0.31 - - [14/Jan/2021:23:39:45 +0000] "POST / HTTP/1.1" 200 0 "" "Falcosidekick" 0/965744
+Deleting pod "alpine" in namespace "default"
+```
+
+## Conclusion
+
+With this really simple example, we only scratched the surface of possibilities, everything is possible now, so don't hesitate to share with us on Slack (https://kubernetes.slack.com #falco)
+) your comments, ideas and successes.
```

Comment:
we should add a comment about contributing to falco. https://github.com/falcosecurity/.github/blob/master/CONTRIBUTING.md